src_kt/java/arcs/storage/StoreInterface.kt

```diff
@@ -67,7 +68,9 @@ data class StoreOptions<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
     val storageKey: StorageKey,
     val existenceCriteria: ExistenceCriteria,
     val type: Type,
-    val mode: StorageMode = StorageMode.Direct,
+    val mode: StorageMode =
+        if (storageKey is ReferenceModeStorageKey) StorageMode.ReferenceMode
+        else StorageMode.Direct,
```

Comment:
I feel like `storageKey is ReferenceModeStorageKey ? StorageMode.ReferenceMode : StorageMode.Direct` is nicer/more traditional, but I'll leave it up to you.

Reply:
there's no such thing as a ternary operator in kotlin, so inline if/else is the way to do the same type of thing

Reply:
TIL, thanks!